File: presentation/src/main/java/android/montreal/devfest/presentation/MainViewModule.java
package android.montreal.devfest.presentation;

import android.content.Context;
import android.montreal.devfest.dagger.ActivityContext;
import android.montreal.devfest.presentation.activities.MainActivity;
import android.montreal.devfest.presentation.navigation.Navigator;
import android.montreal.devfest.presentation.presenters.MainPresenter;
import android.montreal.devfest.presentation.presenters.MainPresenterModule;

import dagger.Binds;
import dagger.Module;

@Module(includes = {MainPresenterModule.class/*, NavigationModule.ViewMore.class*/})
public interface MainViewModule {

    @Binds
    @ActivityContext
    Context context(MainActivity activity);

    @Binds
    MainPresenter.MainView view(MainActivity activity);

    @Binds
    Navigator.ViewMore viewMore(MainActivity activity);
}

File: stuff_presentation/src/main/java/android/montreal/devfest/stuff/presentation/activities/MainViewModule.kt
package android.montreal.devfest.stuff.presentation.activities

import android.content.Context
import android.montreal.devfest.dagger.ActivityContext
import android.montreal.devfest.stuff.presentation.navigation.Navigator
import android.montreal.devfest.stuff.presentation.presenters.MainPresenter
import android.montreal.devfest.stuff.presentation.presenters.MainPresenterModule

import dagger.Binds
import dagger.Module

@Module(includes = [MainPresenterModule::class/*, NavigationModule.ViewMore.class*/])
interface MainViewModule {

    @Binds
    @ActivityContext
    fun context(activity: MainActivity): Context

    @Binds
    fun view(activity: MainActivity): MainPresenter.MainView

    @Binds
    fun viewMore(activity: MainActivity): Navigator.ViewMore
}